### PR TITLE
CI: Automatically add native binaries to releases

### LINKED ISSUE #2865: Build native releases
Currently we only distribute Docker images. While it's easy, having native binary can be handy for some use cases.

### PR DESCRIPTION
**Pull request by Geod24**
It took a lot of sweat and tears, as I was originally trying to automatically promote the tag to a release, and bash just would not have it.

Fixes #2865

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# This workflow will only trigger when a release is created,
+# as this is where the artifacts will be visible.
+# To create a release, use the script `infrastructure/github/release.d`.
+#
+# There is no `repository == bosagora/agora` protection like in other CI scripts,
+# as the artifacts are local to the repository, so there is no access issue,
+# and releases aree not part of the synchronization of a fork with an upstream,
```
Comment (hewison-chris) [inline .github/workflows/release.yaml]:
```suggestion
# and releases are not part of the synchronization of a fork with an upstream,
```

**Review by hewison-chris — APPROVED** [review]
LGTM, just a typo

**Review by linked0 — APPROVED** [review]
LGTM

File: .github/workflows/release.yaml (hunk 1 above)
Comment (Geod24) [reply .github/workflows/release.yaml]: Thanks, fixed

**Pull request merged**